### PR TITLE
Adding `renderers` prop, which allow to specify custom renderers.

### PR DESCRIPTION
**Pull request by domderen**

### CODE REVIEW

File: src/HugeTable.js [hunk 1]
```diff
@@ -21,6 +22,12 @@ export class HugeTable extends React.Component {
       name: React.PropTypes.string,
       type: React.PropTypes.string,
     })),
+    renderers: React.PropTypes.shape(RETURNED_DATA_TYPES.reduce((initial, next) => {
+      return {
+        ...initial,
+        [next]: React.PropTypes.func,
+      };
+    }, {header: React.PropTypes.func})),
```
Comment (paulcuth) [inline src/HugeTable.js]: Should `header` be uppercase?

File: src/HugeTable.js [hunk 2]
```diff
@@ -95,6 +102,32 @@ export class HugeTable extends React.Component {
...
+        width={this.state.columnWidths[schemaItem.name]}
+        cell={(props) => this.cellRenderer({...props, schemaItemName: schemaItem.name})}
+        key={schemaItem.name}
+        isResizable
+      />
+    );
+  }
+
+  renderHeader = (props) => {
+    if(this.props.renderers && this.props.renderers.HEADER && typeof this.props.renderers.HEADER === 'function') {
```
Comment (paulcuth) [inline src/HugeTable.js]: Isn't the last condition implied by the `propTypes` and the fact that it has a value (second condition)?

File: src/HugeTable.js (hunk 1 above)
Comment (domderen) [inline src/HugeTable.js]: correct, fixed.

File: src/HugeTable.js (hunk 2 above)
Comment (domderen) [inline src/HugeTable.js]: `PropTypes` validations are not strong validations, and are totally removed in the production code, so without this check if a renderer wouldn't be a function, this code might still break.

Comment (paulcuth) [inline src/HugeTable.js]: I had no idea they were removed in prop... good to know!

**Pull request merged**